snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/rf2/SnomedRf2ExportRequest.java

```diff
@@ -286,8 +262,8 @@ public ExportResult execute(final RepositoryContext context) {
 
 			final Set<String> visitedComponentEffectiveTimes = newHashSet();
 			
-			final long effectiveTimeStart = startEffectiveTime != null ? startEffectiveTime.getTime() : 0;
-			final long effectiveTimeEnd =  endEffectiveTime != null ? endEffectiveTime.getTime() : Long.MAX_VALUE;
+			final long effectiveTimeStart = startEffectiveTime != null ? EffectiveTimes.getEffectiveTime(startEffectiveTime, DateFormats.SHORT) : 0;
```

Comment:
Please move parsing to `SnomedRf2ExportRequestBuilder#doBuild`, if possible, so that we know that all parameters are valid by the time the request starts.

Reply:
Good idea! 👍